src/Processors/Formats/Impl/ORCBlockInputFormat.cpp

```diff
@@ -180,13 +180,26 @@ std::optional<size_t> ORCSchemaReader::readNumberOrRows()
 
 void registerInputFormatORC(FormatFactory & factory)
 {
-    factory.registerInputFormat(
+    factory.registerRandomAccessInputFormat(
         "ORC",
-        [](ReadBuffer & buf, const Block & sample, const RowInputFormatParams &, const FormatSettings & settings)
+        [](ReadBuffer & buf,
+           const Block & sample,
+           const FormatSettings & settings,
+           const ReadSettings &  read_settings,
+           bool  is_remote_fs,
+           size_t /* max_download_threads */,
+           size_t /* max_parsing_threads */)
         {
             InputFormatPtr res;
             if (settings.orc.use_fast_decoder)
-                res = std::make_shared<NativeORCBlockInputFormat>(buf, sample, settings);
+            {
+                bool has_file_size = isBufferWithFileSize(buf);
```

Comment:
pls add `const` where applicable

Reply:
sorry, you mean adding const to `has_file_size` or `seekable_in`?

Reply:
We should not add const to `seekable_in` because `supportsReadAt` is not a const method.

Reply:
`has_file_size`, `use_prefetch`, `min_bytes_for_seek`